yaml-tests/src/test/java/MultiServerConnectionFactoryTest.java

```diff
@@ -42,61 +43,119 @@ void testDefaultPolicy(int initialConnection) throws SQLException {
         MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
                 MultiServerConnectionFactory.ConnectionSelectionPolicy.DEFAULT,
                 initialConnection,
-                dummyConnectionFactory(),
-                List.of(dummyConnectionFactory()));
+                dummyConnectionFactory("Primary"),
+                List.of(dummyConnectionFactory("Alternate")));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
-        MultiServerConnectionFactory.MultiServerRelationalConnection connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(0, connection.getCurrentConnectionSelector());
+
+        MultiServerConnectionFactory.MultiServerRelationalConnection connection =
+                (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
```

Comment:
Are the assertions about `getCurrentConnectionSelector()` necessary anymore? I think asserting about the created statements is more trustworthy.

Reply:
These can be removed. Done.